src/components/widgets/DateTimeWidget.js

```diff
@@ -0,0 +1,37 @@
+import React, { PropTypes } from "react";
+
+
+function DateTimeWidget({
+  schema,
+  id,
+  placeholder,
+  value,
+  defaultValue,
+  required,
+  onChange
+}) {
+  return (
+    <input type="datetime"
```

Comment:
It looks like the right type is `datetime-local`.
See https://developer.mozilla.org/en/docs/Web/HTML/Element/Input#Attributes and https://github.com/whatwg/html/issues/336